### PR TITLE
Example/hierarchy

### PR DESCRIPTION
**Pull request by kendelljoseph**
## Description

This adds a hierarchy example that demonstrates the capability to create a hierarchy between `entities` and `people`.

- [x] I have read and understand the [CONTRIBUTING.md](https://github.com/payloadcms/payload/blob/main/CONTRIBUTING.md) document in this repository.

## Type of change

- [x] Change to the [examples](https://github.com/payloadcms/payload/tree/main/examples) directory (does not affect core functionality)

## Checklist:

- [x] Existing test suite passes locally with my changes
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by DanRibbens — APPROVED** [review]

**Review by jacobsfletch — CHANGES_REQUESTED** [review]
The README needs to include a high-level description about _what_ the example is demonstrating, in a 2-3 concise sentences. It also needs to include a "How it works" section that outlines the setup and all hooks, etc that are used to achieve its core functionality.

File: examples/hierarchy/README.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# Payload Hierarchy Example
+
+A hierarchy for [Payload](https://github.com/payloadcms/payload).
```
Comment (jacobsfletch) [inline examples/hierarchy/README.md:3]:
```suggestion
This example demonstrates how to achieve a virtual hierarchy between documents in your [Payload](https://github.com/payloadcms/payload) application.```

File: examples/hierarchy/README.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+This repo was created by running `npx create-payload-app@latest` and selecting the "blank" template then it was modified.
```
Comment (jacobsfletch) [inline examples/hierarchy/README.md:5]:
```suggestion
```

File: examples/hierarchy/README.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+See the official [Examples Directory](https://github.com/payloadcms/payload/tree/main/examples) for details on how to use Payload in a variety of different ways.
```
Comment (jacobsfletch) [inline examples/hierarchy/README.md:7]:
```suggestion
```

File: examples/hierarchy/README.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+1. Then `cd YOUR_PROJECT_REPO && cp .env.example .env`
+1. Next `yarn && yarn dev` (or `docker-compose up`, see [Docker](#docker))
+1. Now `open http://localhost:3000/admin` to access the admin panel
+1. Create your first admin user using the form on the page
+
+That's it! Changes made in `./src` will be reflected in your app.
+
+## How it works
+
+This is an example parent-child hierarchy between Entities and People.  Entities use of a virtual field to find people and entities it is related to.
```
Comment (jacobsfletch) [inline examples/hierarchy/README.md:23]: This example achieves parent/child relationships between your documents through the use of virtual fields. When you query a document with the `?children=true` query param, an `afterRead` hook is used to populate the documents within its own tree.

For more information on how virtual fields, see the [Official Virtual Fields Example]([Virtual Fields](https://github.com/payloadcms/payload/tree/main/examples/virtual-fields).

**Review by jacobsfletch — APPROVED** [review]

**Pull request merged**